src/test/rustdoc-json/compare.py

```diff
@@ -39,11 +39,24 @@ def _check_subset(expected, actual, trace):
                 "expected type `{}`, got `{}`".format(expected_type, actual_type), trace
             )
         if expected_type in (str, int, bool) and expected != actual:
-            raise SubsetException("expected `{}`, got: `{}`".format(expected, actual), trace)
+            if expected_type == str and actual.startswith(base_dir):
+                if actual.replace(base_dir + "/", "") != expected:
+                    raise SubsetException(
+                        "expected `{}`, got: `{}`".format(
+                            expected, actual.replace(base_dir + "/", "")
```

Comment:
What is this doing?

Reply:
This is a hack to get around the real output being `"filename": "/home/nixon/upstreams/rust/rust/src/test/rustdoc-json/structs.rs"`, but the test is `"filename": "structs.rs",`.

Long term, it's probably usefull to do something like `$TEST_BUILD_DIR`, like the ui tests have.

Reply:
Ok, can you leave those examples as a comment in the source? Feel free to change the absolute path to something else as long as it's still clear it's absolute and not relative.